### ISSUE TITLE
NodeType configuration with falsy supertype leads to exception

### ISSUE THREAD

**Issue by ger4003** [neos/neos-development-collection#3520]
### Description

Inside `\Neos\Neos\Aspects\NodeTypeConfigurationEnrichmentAspect::setPropertyLabels` the type declaration of `NodeType $superType` leads to an exception, if a superType is declared `false` in nodeType configuration.

**Exception**

```
Argument 2 passed to Neos\Neos\Aspects\NodeTypeConfigurationEnrichmentAspect_Original::Neos\Neos\Aspects\{closure}() must be an instance of Neos\ContentRepository\Domain\Model\NodeType, null given
```

**Example nodeType configuration**:

```yaml
My.Package:Foo.Bar:
  superTypes:
    'Neos.Neos:Content': true
    'My.Package:SomeMixin': false
```

The error was introduced with [BUGFIX: Enable translations for overridden selectbox options](https://github.com/neos/neos-development-collection/commit/ae190f3de36627cacc50c0114444cb955e1a3203)

### Steps to Reproduce

1. Checkout neos/neos:5.3.5
2. Open Neos backend

#### Expected behavior

Neos backend is shown.

#### Actual behavior

Exception is thrown 

```
Argument 2 passed to Neos\Neos\Aspects\NodeTypeConfigurationEnrichmentAspect_Original::Neos\Neos\Aspects\{closure}() must be an instance of Neos\ContentRepository\Domain\Model\NodeType, null given
```

### Affected Versions

Neos: 

- 5.3.5
- 7.0.4
- 7.1.2 
- 7.2.1

**Comment by TheLalaMan** [comment]
Can confirm this on Neos 5.3.5. /cc @grebaldi

**Comment by patricekaufmann** [comment]
I have the same issue occuring when updating from 7.2.0 to 7.2.1. Multiple packages get updates in that command so I can't really track it down further. However the following downgrade fixes the issue: 

```
  - Downgrading neos/content-repository (7.2.1 => 7.2.0)
  - Downgrading neos/media (7.2.1 => 7.2.0)
  - Downgrading neos/media-browser (7.2.1 => 7.2.0)
  - Downgrading neos/neos (7.2.1 => 7.2.0)
  - Downgrading neos/site-kickstarter (7.2.1 => 7.2.0)
```

**Comment by networkteamcom** [comment]
@reflexxion has created a fix #3518 already.

**Comment by kdambekalns** [comment]
@jonnitto I didn't close this, since the fix is neither upmerged nor released yet. 🤷‍♂️
